pkg/cloud/services/compute/instance.go

```diff
@@ -490,16 +500,36 @@ func (s *Service) getBlockDevices(eventObject runtime.Object, instanceSpec *Inst
 	}
 
 	for _, blockDeviceSpec := range instanceSpec.AdditionalBlockDevices {
-		blockDevice, err := s.getOrCreateVolumeBuilder(eventObject, instanceSpec, blockDeviceSpec, "", fmt.Sprintf("Additional block device for %s", instanceSpec.Name))
-		if err != nil {
-			return []bootfromvolume.BlockDevice{}, err
+		var bdUUID string
+		var localDiskSizeGiB int
+		var sourceType bootfromvolume.SourceType
+		var destinationType bootfromvolume.DestinationType
+		if blockDeviceSpec.Name == "root" {
+			return []bootfromvolume.BlockDevice{}, fmt.Errorf("block device name 'root' is reserved")
+		}
```

Comment:
Can you put this in the validating webhook instead?

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/f28da1834f1ea79c0e8e5617aa14607b045ca53b/api/v1alpha7/openstackmachine_webhook.go#L58

Reply:
done